text/001-css-nesting.md

```diff
@@ -0,0 +1,574 @@
+  margin: calc(50px / 2);
+}
+```
+
+## 7. Variants Reference
+Inspired by the [JSS plugin nested](https://cssinjs.org/jss-plugin-nested?v=v10.10.0#use-rulename-to-reference-a-local-rule-within-the-same-style-sheet), this feature can be reference a local rule.
+
+Use the `%` symbol.
+
+**Code:**
+```typescript
+export const myCss = cssVariant({
+  primary: {
+    color: "red",
+    ":has(%secondary)": {
+      color: "blue",
+    }
+  },
+  secondary: {
+    color: "black",
+    "%primary &":{
+      color: "white"
+    }
+  }
+});
+```
+
+**Convert to Vanilla Extract:**
+```typescript
+export const myCss = styleVariant({
+  primary: {
+    color: "red",
+  },
+  secondary: {
+    color: "black",
+  }
+});
+globalCSS(`${myCss["primary"]}:has(${myCss["secondary"]})`, {
+  color: "blue"
+});
+globalCSS(`${myCss["primary"]} ${myCss["secondary"]}`, {
+  color: "white"
+});
+```
+
+**Compiled:**
+```css
+.[FILE_NAME]_myCSS_primary__[HASH] {
+  color: red;
+}
+
+.[FILE_NAME]_myCSS_primary__[HASH]:has(.[FILE_NAME]_myCSS_secondary__[HASH]) {
+  color: blue;
+}
+
+.[FILE_NAME]_myCSS_secondary__[HASH] {
+  color: black;
+}
+
+.[FILE_NAME]_myCSS_primary__[HASH] .[FILE_NAME]_myCSS_secondary__[HASH] {
+  color: white;
+}
+```
+
+# Reference-level explanation
+It's more complicated, but can be implemented almost exactly like [CSS Literals](./000-css-literals.md).
+
+# Drawbacks
+Nested syntax reduces duplication, but if abused, it can be hard to know the context.
+
+# Rationale and alternatives
+The syntax for literal CSS is now on the same level as the preprocessors.
+The selector stuff is definitely [Stylus](https://stylus-lang.com/docs/selectors.html#partial-reference)-rich, but it's generally okay.
+
+## Partial Reference
+
+Using the `^[N]`, `^[N..M]` syntax, You can reference the parent, but it's confusing.
+It's easy to make a mistake, and it's easy to encourage nesting.
+
+If you need them in the future, you should add them very carefully.
+
+**SCSS like Syntax:**
+```scss
+.foo {
+  &__bar {
+    &__baz {
+      width: 10px;
+
+      ^[0]:hover & {
+        width: 20px;
+      }
+      ^[1]:hover & {
+        width: 30px;
+      }
+      ^[2]:hover & {
+        width: 40px;
+      }
+      ^[-1]:hover & {
+        width: 50px;
+      }
+      ^[-2]:hover & {
+        width: 60px;
+      }
+      ^[1..-1]:hover & {
+        width: 70px
+      }
+    }
+  }
+}
+```
+
+**Compiled:**
+```css
+.foo__bar_baz {
+  width: 10px;
+}
+
+.foo:hover .foo__bar_baz {
+  width: 20px;
+}
+
+.foo__bar:hover .foo__bar_baz {
+  width: 30px;
+}
+
+.foo__bar_baz:hover .foo__bar_baz {
+  width: 40px;
+}
+
+.foo__bar:hover .foo__bar_baz {
+  width: 50px;
+}
+
+.foo:hover .foo__bar_baz {
+  width: 60px;
+}
+
+__bar__baz:hover .foo__bar__baz {
+  width: 70px;
+}
+```
+
+## Path-like Reference
+
+Using the `/`, `~`, `../` syntax, which looks like it's used in paths, is less confusing, but still complicated.
+
+**SCSS like Syntax:**
+```scss
+.block {
+  &__element {
```

Comment:
it's also basic syntax for `Path-like Reference`?

Reply:
No. `&` is a `Nested Selectors` syntax.